src/components/ha-selector/ha-selector-datetime.ts

```diff
@@ -39,13 +39,13 @@ export class HaDateTimeSelector extends LitElement {
           .locale=${this.hass.locale}
           .disabled=${this.disabled}
           .required=${this.required}
-          .value=${values?.[0]}
+          .value=${values?.[0] === "undefined" ? undefined : values?.[0]}
           @value-changed=${this._valueChanged}
         >
         </ha-date-input>
         <ha-time-input
           enable-second
-          .value=${values?.[1] || "0:00:00"}
+          .value=${values?.[1] || "00:00:00"}
```

Comment:
If pass "0:00:00" to time-input, it reformats it as "00:00:00" and fires an unwanted value-changed event.